apps/testing/src/main/java/com/akto/testing/Main.java

```diff
@@ -95,88 +103,91 @@ public static void main(String[] args) throws InterruptedException {
+                Bson filter1 = Filters.eq(TestingRun.STATE, TestingRun.State.SCHEDULED);
+                Bson filter2 = Filters.and(
+                        Filters.eq(TestingRun.STATE, TestingRun.State.RUNNING),
+                        Filters.lte(TestingRun.SCHEDULE_TIMESTAMP, delta)
+                );
 
-            Bson update = Updates.combine(
-                    Updates.set(TestingRun.PICKED_UP_TIMESTAMP, Context.now()),
-                    Updates.set(TestingRun.STATE, TestingRun.State.RUNNING)
-            );
+                Bson update = Updates.combine(
+                        Updates.set(TestingRun.PICKED_UP_TIMESTAMP, Context.now()),
+                        Updates.set(TestingRun.STATE, TestingRun.State.RUNNING)
+                );
 
-            TestingRun testingRun = TestingRunDao.instance.getMCollection().findOneAndUpdate(
-                    Filters.or(filter1,filter2), update);
+                int start = Context.now();
 
+                TestingRun testingRun = TestingRunDao.instance.getMCollection().findOneAndUpdate(
+                        Filters.or(filter1,filter2), update);
 
-            if (testingRun == null) {
-                try {
-                    Thread.sleep(10 * 1000L);
-                    continue;
-                } catch (InterruptedException e) {
-                    throw new RuntimeException(e);
+                if (testingRun == null) {
+                    return;
                 }
-            }
 
-            long timestamp = testingRun.getId().getTimestamp();
-            long seconds = Context.now() - timestamp;
-            loggerMaker.infoAndAddToDb("Found one + " + testingRun.getId().toHexString() + " created: " + seconds + " seconds ago", LogDb.TESTING);
-            if (testingRun.getTestIdConfig() > 1) {
-                TestingRunConfig testingRunConfig = TestingRunConfigDao.instance.findOne(Constants.ID, testingRun.getTestIdConfig());
-                if (testingRunConfig != null) {
-                    loggerMaker.infoAndAddToDb("Found testing run config with id :" + testingRunConfig.getId(), LogDb.TESTING);
-                    testingRun.setTestingRunConfig(testingRunConfig);
-                } else {
-                    loggerMaker.errorAndAddToDb("Couldn't find testing run config id for " + testingRun.getTestIdConfig(), LogDb.TESTING);
-                }
-            }
+                loggerMaker.infoAndAddToDb("Found one + " + testingRun.getId().toHexString(), LogDb.TESTING);
+
+                try {
+                    AccountSettings accountSettings = AccountSettingsDao.instance.findOne(new BasicDBObject());
+                    boolean runStatusCodeAnalyser = accountSettings == null ||
+                            accountSettings.getSetupType() != AccountSettings.SetupType.PROD;
+
+                    SampleMessageStore sampleMessageStore = SampleMessageStore.create();
+                    AuthMechanismStore authMechanismStore = AuthMechanismStore.create();
```

Comment:
Not sure if we need to bring sample message fetch etc here. Previous code should work imo, but we can check once

Reply:
Also main file shouldn't have a lot of logic, gets harder to maintain later on